### PR TITLE
Added fromYaml(...) method for aprilgrid config.

### PR DESCRIPTION
**Pull request by fabianbl**
Loading aprilgrid config from a yaml file (same configuration as in Kalibr).

### CODE REVIEW

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 1]
```diff
@@ -16,6 +17,32 @@
...
+    CHECK_GT(target_config.tag_size_meter, 0.0);
+    double relative_tag_spacing;
+    YAML::safeGet(yaml_node, "tagSpacing", &relative_tag_spacing);
+    target_config.tag_inbetween_space_meter =
+        relative_tag_spacing * target_config.tag_size_meter;
+    CHECK_GT(target_config.tag_inbetween_space_meter, 0.0);
+  } catch (const std::exception& ex) {
+    LOG(FATAL) << "Failed to load yaml file " << yaml_file
+               << " with the error: \n " << ex.what() << ".";
+  }
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: [catch(...)](https://github.com/ethz-asl/siemens_tools/pull/25) for other exceptions?

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 2]
```diff
@@ -16,6 +17,32 @@
 namespace aslam {
 namespace calibration {
 
+TargetAprilGrid::TargetConfiguration
+TargetAprilGrid::TargetConfiguration::fromYaml(const std::string& yaml_file) {
+  TargetConfiguration target_config;
+  try {
+    YAML::Node yaml_node = YAML::LoadFile(yaml_file.c_str());
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: const

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 3]
```diff
@@ -16,6 +17,32 @@
...
+    std::string target_type;
+    YAML::safeGet(yaml_node, "target_type", &target_type);
+    CHECK_EQ(target_type, "aprilgrid");
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: Maybe more explanatory exception?

File: aslam_cv_calibration/include/aslam/calibration/target-aprilgrid.h [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 #include <memory>
 #include <vector>
+#include <string>
```
Comment (eggerk) [inline aslam_cv_calibration/include/aslam/calibration/target-aprilgrid.h]: alph

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 5]
```diff
@@ -2,6 +2,7 @@
 #include <memory>
 #include <vector>
 
+#include <aslam/common/yaml-serialization.h>
```
Comment (eggerk) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: alph

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 6]
```diff
@@ -16,6 +17,32 @@
...
+    YAML::safeGet(yaml_node, "tagRows", &target_config.num_tag_rows);
+    CHECK_GT(target_config.num_tag_rows, 0u);
+    YAML::safeGet(yaml_node, "tagSize", &target_config.tag_size_meter);
+    CHECK_GT(target_config.tag_size_meter, 0.0);
+    double relative_tag_spacing;
+    YAML::safeGet(yaml_node, "tagSpacing", &relative_tag_spacing);
+    target_config.tag_inbetween_space_meter =
+        relative_tag_spacing * target_config.tag_size_meter;
+    CHECK_GT(target_config.tag_inbetween_space_meter, 0.0);
+  } catch (const std::exception& ex) {
```
Comment (eggerk) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: only catch yaml exception?

File: aslam_cv_calibration/src/target-aprilgrid.cc [hunk 7]
```diff
@@ -36,7 +36,7 @@ TargetAprilGrid::TargetConfiguration::fromYaml(const std::string& yaml_file) {
     target_config.tag_inbetween_space_meter =
         relative_tag_spacing * target_config.tag_size_meter;
     CHECK_GT(target_config.tag_inbetween_space_meter, 0.0);
-  } catch (const std::exception& ex) {
+  } catch (const YAML::Exception& ex) {
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-aprilgrid.cc]: what happens to all other exceptions?

**Comment by fabianbl** [comment]
test this please

File: aslam_cv_calibration/src/target-aprilgrid.cc (hunk 7 above)
Comment (fabianbl) [reply aslam_cv_calibration/src/target-aprilgrid.cc]: @floriantschopp I guess there should not be any other exceptions than the one of not being able to read the yaml file?

Comment (floriantschopp) [reply aslam_cv_calibration/src/target-aprilgrid.cc]: Agree. But in the case there is some, is it automatically thrown or do you have to do it manually by 
```cpp
catch(...) {
throw();
}
```
I'm not sure tbh.

**Comment by fabianbl** [comment]
test this please (problem downloading eigen...)

File: aslam_cv_calibration/src/target-aprilgrid.cc (hunk 7 above)
Comment (fabianbl) [reply aslam_cv_calibration/src/target-aprilgrid.cc]: I guess the exception will be thrown by ``YAML::LoadFile` [here](https://github.com/jbeder/yaml-cpp/blob/master/src/parse.cpp#L35).

Comment (eggerk) [reply aslam_cv_calibration/src/target-aprilgrid.cc]: @floriantschopp The program should terminate if an exception is thrown that you don't catch. This should be fine here as we don't expect any other exceptions to be thrown.

Comment (floriantschopp) [reply aslam_cv_calibration/src/target-aprilgrid.cc]: ok makes sense, thank you!

**Pull request merged**